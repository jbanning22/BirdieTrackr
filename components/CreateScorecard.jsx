import {
  StyleSheet,
  Text,
  View,
  Button,
  TextInput,
  TouchableOpacity,
  KeyboardAvoidingView,
} from 'react-native';
import React, {useState} from 'react';
import axios from 'axios';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {SafeAreaView} from 'react-native-safe-area-context';

const CreateScorecard = ({navigation}) => {
  const [courseLength, setCourseLength] = useState(0);
  const [courseName, setCourseName] = useState('');
  const [active9Button, setActive9Button] = useState(false);
  const [active18Button, setActive18Button] = useState(false);
  const [scorecardData, setScorecardData] = useState({});

  function setLength9() {
    setCourseLength(9);
    setActive9Button(true);
    setActive18Button(false);
  }
  function setLength18() {
    setCourseLength(18);
    setActive18Button(true);
    setActive9Button(false);
  }

  const createScorecard = async () => {
    const token = await AsyncStorage.getItem('token');
    const headers = {
      Authorization: `Bearer ${token}`,
    };
    try {
      const scorecard = await axios.post(
        'http://192.168.1.154:3000/scorecard',
        {courseLength: courseLength, courseName: courseName},
        {headers},
      );
      navigation.navigate('FullScorecard', {
        id: scorecard.data.id,
      });
      return setScorecardData(scorecard.data);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <SafeAreaView style={styles.box1}>
      <Text style={styles.questionText}>Name of the course?</Text>
      <KeyboardAvoidingView>
        <TextInput
          placeholder="Course Name"
          style={styles.loginTextInput}
          value={courseName}
          onChangeText={setCourseName}
        />
      </KeyboardAvoidingView>
      <Text style={styles.questionText}>How Many Holes?</Text>
      <View style={{flexDirection: 'row'}}>
        <TouchableOpacity
          style={active9Button ? styles.lengthButton2 : styles.lengthButton1}
          activeOpacity={0.3}
          onPress={setLength9}>
          <Text style={styles.buttonText}>9</Text>
        </TouchableOpacity>
        <Text style={styles.orText}>Or</Text>
        <TouchableOpacity
          style={active18Button ? styles.lengthButton2 : styles.lengthButton1}
          activeOpacity={0.3}
          onPress={setLength18}>
          <Text style={styles.buttonText}>18</Text>
        </TouchableOpacity>
      </View>
      <TouchableOpacity
        style={styles.createScorecardButton}
        onPress={createScorecard}>
        <Text style={styles.createScorecardText}>Create Scorecard</Text>
      </TouchableOpacity>
      <View style={styles.box2}>
        <TouchableOpacity
          style={styles.backButton}
          onPress={() => navigation.navigate('Scorecard')}>
          <Text style={styles.backButtonText}>Go Back</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

export default CreateScorecard;

const styles = StyleSheet.create({
  box1: {
    flex: 1,
    alignItems: 'center',
    alignContent: 'center',
    justifyContent: 'center',
  },
  box2: {
    justifyContent: 'center',
    alignContent: 'center',
  },
  questionText: {
    fontSize: 20,
    fontWeight: '400',
    fontFamily: 'Helvetica',
    marginBottom: 20,
    marginTop: 20,
  },
  loginTextInput: {
    height: 50,
    width: 200,
    padding: 10,
    backgroundColor: 'white',
    marginTop: 10,
  },
  createScorecardButton: {
    justifyContent: 'center',
    alignContent: 'center',
    height: 60,
    width: 200,
    backgroundColor: '#52BEDB',
    padding: 10,
    marginTop: 80,
    borderRadius: 10,
  },
  createScorecardText: {
    fontSize: 20,
    fontWeight: '600',
    alignSelf: 'center',
    color: 'white',
  },
  lengthButton1: {
    width: 50,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#DB6F52',
    margin: 35,
    marginBottom: 15,
    borderRadius: 25,
  },
  lengthButton2: {
    width: 50,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#52BEDB',
    margin: 35,
    marginBottom: 15,
    borderRadius: 25,
  },
  backButton: {
    width: 80,
    height: 40,
    padding: 5,
    marginTop: 50,
  },
  buttonText: {
    alignSelf: 'center',
    alignItems: 'center',
    color: 'white',
    fontWeight: '400',
    fontSize: 22,
  },
  backButtonText: {
    color: 'black',
    alignSelf: 'center',
    alignItems: 'center',
  },
  orText: {
    alignSelf: 'center',
    alignItems: 'center',
    color: 'black',
    fontWeight: '400',
    fontSize: 25,
    marginTop: 18,
  },
});
