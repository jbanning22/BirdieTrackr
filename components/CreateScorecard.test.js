import React from 'react';
import {render, screen} from '@testing-library/react-native';
import CreateScorecard from './CreateScorecard';
import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
import axios from 'axios';

jest.mock('axios');
jest.mock('@fortawesome/react-native-fontawesome', () => ({
  FontAwesomeIcon: () => null,
}));

describe('Create Scorecard Screen', () => {
  let queryClient;
  beforeEach(() => {
    queryClient = new QueryClient();
    queryClient.setDefaultOptions({
      queries: {cacheTime: 0},
      retry: false,
      refetchOnWindowFocus: false,
    });
    axios.post.mockResolvedValue({
      data: {
        id: 1,
      },
    });
  });
  afterEach(() => {
    queryClient.clear();
    jest.clearAllMocks();
  });
  it('should match the snapshot', async () => {
    const {toJSON} = render(
      <QueryClientProvider client={queryClient}>
        <CreateScorecard />
      </QueryClientProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
  it('should render question title', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <CreateScorecard />
      </QueryClientProvider>,
    );
    expect(screen.getByText('Where are you playing?')).toBeTruthy();
  });
  it('should render question on course name', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <CreateScorecard />
      </QueryClientProvider>,
    );
    expect(screen.getByText('Name of the Course')).toBeTruthy();
  });
  it('should be able to input course name', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <CreateScorecard />
      </QueryClientProvider>,
    );
    expect(screen.getByPlaceholderText('Course Name')).toBeTruthy();
  });
  it('should render question on course length', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <CreateScorecard />
      </QueryClientProvider>,
    );
    expect(screen.getByText('How Many Holes?')).toBeTruthy();
  });
  it('should be able to input course length', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <CreateScorecard />
      </QueryClientProvider>,
    );
    expect(screen.getByText('9')).toBeTruthy();
    expect(screen.getByText('18')).toBeTruthy();
  });
  it('should render button to create scorecard', async () => {
    render(
      <QueryClientProvider client={queryClient}>
        <CreateScorecard />
      </QueryClientProvider>,
    );
    expect(screen.getByText('Create Scorecard')).toBeTruthy();
  });
});
