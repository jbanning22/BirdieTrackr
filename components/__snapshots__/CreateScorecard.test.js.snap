// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Scorecard Screen should match the snapshot 1`] = `
<RNCSafeAreaView
  edges={
    [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "flex": 1,
        "justifyContent": "space-between",
        "resizeMode": "cover",
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../assets/images/BasketBackground2.png",
        }
      }
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": undefined,
            "width": undefined,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "flexDirection": "row",
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      />
      <Text
        style={
          {
            "flex": 1,
            "fontFamily": "Satoshi-Medium",
            "fontSize": 24,
            "fontWeight": "400",
            "margin": 6,
            "marginLeft": 45,
          }
        }
      >
        Where are you playing?
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "#F9FAFB",
          "borderRadius": 15,
          "margin": 20,
        }
      }
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#909090",
            "fontFamily": "Satoshi-Medium",
            "fontSize": 18,
            "fontWeight": "400",
            "marginBottom": 10,
            "marginTop": 20,
          }
        }
      >
        Name of the Course
      </Text>
      <View
        onLayout={[Function]}
      >
        <TextInput
          onChangeText={[Function]}
          placeholder="Course Name"
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "white",
              "borderRadius": 10,
              "height": 50,
              "marginTop": 10,
              "padding": 10,
              "textAlign": "center",
              "width": 300,
            }
          }
          value=""
        />
      </View>
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#909090",
            "fontFamily": "Satoshi-Medium",
            "fontSize": 18,
            "fontWeight": "400",
            "marginBottom": 10,
            "marginTop": 20,
          }
        }
      >
        How Many Holes?
      </Text>
      <View
        style={
          {
            "flexDirection": "row",
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#9CA3AF",
              "borderRadius": 8,
              "height": 32,
              "justifyContent": "center",
              "margin": 35,
              "opacity": 1,
              "width": 104,
            }
          }
        >
          <Text
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "color": "white",
                "fontSize": 22,
                "fontWeight": "400",
              }
            }
          >
            9
          </Text>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#9CA3AF",
              "borderRadius": 8,
              "height": 32,
              "justifyContent": "center",
              "margin": 35,
              "opacity": 1,
              "width": 104,
            }
          }
        >
          <Text
            style={
              {
                "alignItems": "center",
                "alignSelf": "center",
                "color": "white",
                "fontSize": 22,
                "fontWeight": "400",
              }
            }
          >
            18
          </Text>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#2D6061",
          "borderRadius": 14,
          "height": 56,
          "justifyContent": "center",
          "opacity": 1,
          "width": 327,
        }
      }
    >
      <Text
        style={
          {
            "color": "white",
            "fontFamily": "Satoshi-Medium",
            "fontSize": 16,
            "fontWeight": "500",
          }
        }
      >
        Create Scorecard
      </Text>
    </View>
  </View>
</RNCSafeAreaView>
`;
